### ISSUE TITLE
Scrolling is broken when a custom onServerSearch function Is defined

### ISSUE THREAD

**Issue by MichalLauer** [sa-si-dev/virtual-select#319]
Hello,

first of all, thanks for this great package!
I'm trying to implement a custom onServerSearch function, but when a user searches for a specific term, the scroll-down gets broken and all options dissapear. A minimal reprex is here:

```js
 function searchLabel(searchValue, virtualSelect) {
   const searchWords = searchValue.split(/[\s\/]/);
   let grouped_data = [];
   virtualSelect.options.forEach(item => {
     item.isVisible = searchWords.every(word => item.label.match(new RegExp(`(?<!\S)${word}`, "giu")) !== null);
     grouped_data.push(item);
   });
   virtualSelect.setServerOptions(grouped_data);
 }
 VirtualSelect.init({
   ele: "#pick",
   multiple: true,
   search: true,
   keepAlwaysOpen: true,
   onServerSearch: searchLabel,
   options: [{ label: "a" },
             { label: "n" },
             { label: "u" },
             { label: "v" },
             { label: "w" },
             { label: "x" },
             { label: "y" },
             { label: "k" },
             { label: "p" },
             { label: "za" }
            ]
 });
```
If you try to search for "a" and scroll down, the selection breaks. I've been trying to figure this out for the past week but can't seem to get to the bottom of it.

Any idea why this might cause an issue?
Thanks!

**Comment by MichalLauer** [comment]
After a bit of searching, I think this issue could be related to #318 and/or #297.

In the meantine, I also created a [codepen reprex](https://codepen.io/MichalLauer/pen/VwNyQwz) which might help further narrow down the issue :)

![APenbyMichalLauerBrave2024-04-0316-15-59-ezgif com-video-to-gif-converter](https://github.com/sa-si-dev/virtual-select/assets/20439350/afe32bfb-265d-4f17-b6b6-bc4d3f33de39)

**Comment by gnbm** [comment]
@sa-si-dev Could you help on this one?

**Comment by MichalLauer** [comment]
So the issue is that there is no call to `virtualSelect.setVisibleOptionsCount(...);` If this is manually included, there is no longer any issue - see here https://codepen.io/MichalLauer/pen/RwOqWgr.

Would it be possible to fix it somewhere in the package? I am not very familiar with JS project structure so I'm not sure where to include it.

**Comment by gnbm** [comment]
Hi @MichalLauer 
I was having a similar issue so I decided to take a look at the code and try to fix this.
Created a draft [PR](https://github.com/sa-si-dev/virtual-select/pull/325) with a possible solution, could you please help validate this before opening the PR itself (I'm just having some issues running the tests, where 3 are failing, even though I can't manually get any error but would like double check to mitigate the risk of introducing any regression issue)?
You'll need to get my branch, build it and use for instance the Dev Tools overrides.

**Comment by gnbm** [comment]
Hi @MichalLauer 
Did you get the chance to test the code from my draft [PR](https://github.com/sa-si-dev/virtual-select/pull/325)?

**Comment by MichalLauer** [comment]
Hey,
I tried to build the branch yesterday but could not run any `npm run` or `npm install` commands because of authentication. I tried to create an NPM account, but that did not help either. I want to try again on Thursday. Do you have an idea what might have caused it? I tried googling, but nothing helped.

I am not primarily a JS developer, so I'm not very fluent with npm - I just know the basic stuff.

**Comment by gnbm** [comment]
> Hey, I tried to build the branch yesterday but could not run any `npm run` or `npm install` commands because of authentication. I tried to create an NPM account, but that did not help either. I want to try again on Thursday. Do you have an idea what might have caused it? I tried googling, but nothing helped.
> 
> I am not primarily a JS developer, so I'm not very fluent with npm - I just know the basic stuff.
Hi @MichalLauer 
I didn't have to do anything like that.
You should be able to run `npm install` and after that `npm run build` to generate the `dist` files

**Comment by gnbm** [comment]
Fixed in release [v1.0.43](https://github.com/sa-si-dev/virtual-select/releases/tag/v1.0.43).

**Comment by gnbm** [comment]
Hello @MichalLauer 
The problem you're reporting is not related to the one you confirmed is fixed, so it should be treated separately.
Being a custom regex and a custom function I would first debug your code to understand if the issue is really in Virtual Select itself.
